### PR TITLE
Update Beats to use go 1.11.2

### PR DESCRIPTION
**Pull request by vjsamuel**

### CODE REVIEW

**Comment by ruflin** [comment]
One change worth mentioning here is that now MacOS 10.10 at least is required:

> As announced in the Go 1.10 release notes, Go 1.11 now requires OpenBSD 6.2 or later, macOS 10.10 Yosemite or later, or Windows 7 or later; support for previous versions of these operating systems has been removed.

It's not a problem because our support matrix already has 10.13 in and for APM-server 10.12 : https://www.elastic.co/support/matrix @elastic/apm-server We should probably bring these number in sync.

@elastic/beats Anyone seeing any blockers to jump to 1.11 on master?
@vjsamuel Thanks for tackling this.

**Comment by ph** [comment]
I do not see a blocker to move to 1.11.1, we might see a few hiccups in the tooling but that should not stop us to live on the newest version.

**Comment by ph** [comment]
@vjsamuel I am confused by the commits in that PR, I see a lot of log statement changes?

**Comment by graphaelli** [comment]
Thanks @ruflin, I've requested that change to the compat matrix.

**Comment by graphaelli** [comment]
This actually uses go 1.11.1 for package builds too - the title of this change should be updated accordingly.  Also, we'll need `beats-dev/golang-crossbuild:1.11.1*` to support that, before this is merged.  Finally, all of the `FROM golang:1.10.3` in the various Dockerfiles should be updated as well.

Otherwise, +1 to moving to 1.11.1 while it's early in our dev cycle.

**Comment by vjsamuel** [comment]
I have raised https://github.com/elastic/fsevents/pull/4 to fix Auditbeat for OSX

**Comment by ruflin** [comment]
I assume that is the `golang-crossbuild` repo? https://github.com/elastic/golang-crossbuild

**Comment by andrewkroh** [comment]
I pushed golang-crossbuilder images for Go 1.11.1.

**Comment by ruflin** [comment]
I wanted to compare it with the 1.10.3 PR (https://github.com/elastic/beats/pull/7313) as I remember we always forgot to change the hash when updating which only affects packaging. But it seems this one is gone. Will try to run the packaging locally to see if it passes.

**Comment by ruflin** [comment]
`make snapshot` failed on my machine with the following errors:

```
>> golangCrossBuild: Building for linux/amd64
>> Building using: cmd='build/mage-linux-amd64 golangCrossBuild', env=[CC=gcc, CXX=g++, GOARCH=amd64, GOARM=, GOOS=linux, PLATFORM_ID=linux-amd64]
>> golangCrossBuild: Building for windows/386
>> Building using: cmd='build/mage-linux-amd64 golangCrossBuild', env=[CC=i686-w64-mingw32-gcc, CXX=i686-w64-mingw32-g++, GOARCH=386, GOARM=, GOOS=windows, PLATFORM_ID=windows-386]
# crypto/x509
osxcross: error: cannot find libc++ headers
osxcross: error: while detecting target
>> golangCrossBuild: Building for linux/386
>> Building using: cmd='build/mage-linux-amd64 golangCrossBuild', env=[CC=gcc, CXX=g++, GOARCH=386, GOARM=, GOOS=linux, PLATFORM_ID=linux-386]
Error: running "go build -o build/golang-crossbuild/auditbeat-darwin-amd64 -ldflags -X github.com/elastic/beats/libbeat/version.buildTime=2018-10-30T10:28:42Z -X github.com/elastic/beats/libbeat/version.commit=1dbc7b742c5812b599422fd6a72d269da6e99af6" failed with exit code 2
Error: failed building for darwin/amd64: exit status 2
failed building for darwin/amd64: exit status 2
package ran for 7m20.349946313s
Error: running "docker run --env EXEC_UID=501 --env EXEC_GID=20 --rm --env MAGEFILE_VERBOSE= --env MAGEFILE_TIMEOUT= -v /Users/ruflin/Dev/gopath/src/github.com/elastic/beats:/go/src/github.com/elastic/beats -w /go/src/github.com/elastic/beats/x-pack/auditbeat docker.elastic.co/beats-dev/golang-crossbuild:1.11.1-darwin --build-cmd ../../auditbeat/build/mage-linux-amd64 golangCrossBuild -p darwin/amd64" failed with exit code 1
running "docker run --env EXEC_UID=501 --env EXEC_GID=20 --rm --env MAGEFILE_VERBOSE= --env MAGEFILE_TIMEOUT= -v /Users/ruflin/Dev/gopath/src/github.com/elastic/beats:/go/src/github.com/elastic/beats -w /go/src/github.com/elastic/beats/auditbeat docker.elastic.co/beats-dev/golang-crossbuild:1.11.1-darwin --build-cmd build/mage-linux-amd64 golangCrossBuild -p darwin/amd64" failed with exit code 1
make[2]: *** [release] Error 1
make[1]: *** [release] Error 1
make: *** [snapshot] Error 2
```

@andrewkroh Is this expected to fail?

**Comment by graphaelli** [comment]
This appears to be related to Go 1.11 minimum mac os requirements.  Testing a fix where `OSX_VERSION_MIN` is bumped to `10.10` [here](https://github.com/elastic/golang-crossbuild/blob/094dde9ce2e8519ef0fcdb38959901d1cfbcb018/go1.11/darwin/Dockerfile#L16)

**Comment by graphaelli** [comment]
The solution is a bit more involved than that - opened https://github.com/elastic/golang-crossbuild/issues/6 and expect we can move this change forward once that one is resolved and new images are available.

**Comment by vjsamuel** [comment]
Not sure if the failure in metricbeat is related to this PR. Can someone trigger the build again?

**Comment by vjsamuel** [comment]
looks like go1.11.2 is out. https://github.com/golang/go/issues?q=milestone%3AGo1.11.2

Should we move to that?

**Comment by urso** [comment]
Either go1.11.2 or go1.10.5 :)

I know it's tempting to update, but I wonder if we want to stay compatible with gccgo (which still supports only go1.10).

**Comment by vjsamuel** [comment]
@urso 
Go 1.11 for gccgo should be out in 8.3 which is probably around the next month or so. The feature is already in trunk.

https://gcc.gnu.org/ml/gcc-cvs/2018-10/msg00206.html

**Comment by vjsamuel** [comment]
@urso @graphaelli i have moved to 1.11.2

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -57,6 +57,7 @@ https://github.com/elastic/beats/compare/v6.4.0...master[Check the HEAD diff]
 - Fix bug in loading dashboards from zip file. {issue}8051[8051]
 - The export config subcommand should not display real value for field reference. {pull}xxx[xxx]
 - The export config subcommand should not display real value for field reference. {pull}8769[8769]
+- Update Beats to use go 1.11.2 {pull}8746[8746]
```
Comment (urso) [inline CHANGELOG.asciidoc]: This changelog is mostly for users, who do not care about compiler versions :)

Let's add the entry to the developers changelog.

**Comment by graphaelli** [comment]
Changes here lgtm, thanks for doing this @vjsamuel.  https://github.com/elastic/golang-crossbuild/pull/8 is up for the required packaging updates.

File: filebeat/autodiscover/builder/hints/logs.go [hunk 2]
```diff
@@ -185,7 +185,7 @@ func (l *logHints) getFilesets(hints common.MapStr, module string) map[string]*f
 
 	moduleFilesets, err := l.Registry.ModuleFilesets(module)
 	if err != nil {
-		logp.Err("Error retrieving module filesets", err)
+		logp.Err("Error retrieving module filesets: %v", err)
```
Comment (urso) [inline filebeat/autodiscover/builder/hints/logs.go]: We should use `%+v` for errors.

File: libbeat/autodiscover/appenders/config/config.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ func NewConfigAppender(cfg *common.Config) (autodiscover.Appender, error) {
 		if conf.ConditionConfig != nil {
 			cond, err = conditions.NewCondition(conf.ConditionConfig)
 			if err != nil {
-				logp.Warn("config", "unable to create condition due to error: %v", err)
+				logp.Warn("unable to create condition due to error: %v", err)
```
Comment (urso) [inline libbeat/autodiscover/appenders/config/config.go]: error -> `%+v`

**Comment by vjsamuel** [comment]
@urso moved as many errors to `%+v`. the code base has many more :)

**Comment by ruflin** [comment]
jenkins, test this

**Comment by urso** [comment]
Jenkins, please test this.

**Comment by ruflin** [comment]
jenkins, test this

**Comment by ruflin** [comment]
jenkins, test this

**Comment by ruflin** [comment]
jenkins, test this

**Comment by urso** [comment]
@vjsamuel   can you please rebase? There are potential merge conflicts.

I also have had some problems with mage recently when using go1.11.2 (using go1.10.3 did work). Problem was in 6.5 branch, but I'd like to be sure master works.

**Comment by urso** [comment]
Jenkins, please test this.

**Comment by urso** [comment]
Jenkins, test this!

**Comment by ruflin** [comment]
jenkins, test this

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (ruflin) [reply CHANGELOG.asciidoc]: @vjsamuel Can you clean up the changelog?

File: libbeat/docs/version.asciidoc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 :stack-version: 7.0.0-alpha1
 :doc-branch: master
-:go-version: 1.10.3
-:release-state: prerelease
+:go-version: 1.11.2
+:release-state: unreleased
```
Comment (ruflin) [inline libbeat/docs/version.asciidoc]: This worries me. Why did this change? This line should be reverted.

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (urso) [reply CHANGELOG.asciidoc]: I don't think we need it in the CHANGELOG.asciidoc. The developers changelog should be good enough

File: libbeat/docs/version.asciidoc (hunk 4 above)
Comment (urso) [reply libbeat/docs/version.asciidoc]: I guess rebase did opt for the old one. It must say `prerelease`

**Comment by ruflin** [comment]
jenkins, test this

**Comment by ruflin** [comment]
@elastic/beats @elastic/apm-server You should probably update your local environment.

@vjsamuel Thanks for keep pushing this one.

**Pull request merged**